### ISSUE TITLE
Unable to refresh token with Web app

### ISSUE THREAD

**Issue by dramich** [Azure/go-autorest#280]
With a App registration with the type `Web app` I get the following error message from the API when attempting a 2nd refresh of a token:

```
adal: Refresh request failed. Status Code = '401'. Response body: {"error":"invalid_client","error_description":"AADSTS70002: The request body must contain the following parameter: 'client_secret or client_assertion'
```

Step to reproduce:
```
spt, err := adal.NewServicePrincipalTokenFromAuthorizationCode(
	*oauthConfig,
	applicationID,
	clientSecret,
	authorizationCode,
	redirectURI,
	resource,
	callback,
)

if err != nil {
	fmt.Println(err)
	os.Exit(1)
}

// The refresh is required, call above just creates the struct
err = spt.Refresh()
if err != nil {
	fmt.Println(err)
	os.Exit(1)
}

// This call errors
err = spt.Refresh()
if err != nil {
	fmt.Println(err)
	os.Exit(1)
}
```

It appears the issue is coming from [here](https://github.com/Azure/go-autorest/blob/master/autorest/adal/token.go#L628).

Is this me using the client incorrectly?

**Comment by jhendrixMSFT** [comment]
Thanks for reporting this issue, this is a bug in our refresh logic.  I hope to have a fix soon.

**Comment by jhendrixMSFT** [comment]
This is fixed in go-autorest 10.9.2.  Thanks for the bug report!  https://github.com/Azure/go-autorest/releases/tag/v10.9.2
